diff -urN gphpedit-0.9.91.orig/gphpedit.desktop.in gphpedit-0.9.91/gphpedit.desktop.in
--- gphpedit-0.9.91.orig/gphpedit.desktop.in	2006-05-17 01:40:02.000000000 +0200
+++ gphpedit-0.9.91/gphpedit.desktop.in	2007-04-18 18:30:38.889186184 +0200
@@ -3,9 +3,11 @@
 Name=gPHPEdit
 GenericName=PHP/HTML/CSS editor
 Comment=Edit PHP source files
+Comment[pl]=Edycja plików źródłowych PHP
 Exec=gphpedit
 Icon=gphpedit.png
 Terminal=false
 Type=Application
-Categories=Development;IDE;
+Categories=GTK;GNOME;Development;
 Version=0.9.4
+# vi: encoding=utf-8
